staging/src/k8s.io/kubelet/pkg/cri/streaming/portforward/httpstream.go

```diff
@@ -252,11 +261,34 @@ func (h *httpStreamHandler) portForward(p *httpStreamPair) {
 	err := h.forwarder.PortForward(ctx, h.pod, h.uid, int32(port), p.dataStream)
 	klog.V(5).InfoS("Connection request done invoking forwarder.PortForward for port", "connection", h.conn, "request", p.requestID, "port", portString)
 
-	if err != nil {
-		msg := fmt.Errorf("error forwarding port %d to pod %s, uid %v: %v", port, h.pod, h.uid, err)
-		utilruntime.HandleError(msg)
-		fmt.Fprint(p.errorStream, msg.Error())
+	// happy path, we have successfully completed forwarding task
+	if err == nil {
+		return
+	}
+
+	if errors.Is(err, syscall.EPIPE) || errors.Is(err, syscall.ECONNRESET) {
+		// In the process of forwarding, we encountered error types that can be handled:
```

Comment:
What does "can be handled" mean, exactly?

it looks like this change avoids telling the client an error occurred in this case... am I reading that right, and is that the intent? it's not clear why masking the existence of an error to the client is good or desired.

Doesn't handling it this way mean something on the client side would not be informed there was an error? Can't that cause problems on the client side?

Reply:
Separately from whether we call Reset() or Close(), it's not clear to me why we don't write this error to `errorStream`

Reply:
This is still outstanding... seems like we should try to tell the client there was an error. The questions I had on the tests triggering error conditions but not seeing error responses is outstanding as well

Reply:
I think I understand what you are trying to do now. Currently, if we write *any* error back to the errorStream (which is per-subrequest), client-go currently reacts by tearing down the entire portforward connection here:

https://github.com/kubernetes/kubernetes/blob/d831b13e6f6fb5efb566100286190fedca6dd340/staging/src/k8s.io/client-go/tools/portforward/portforward.go#L410-L415

If we want specific errors on specific subrequests to allow the overall portforward operation to continue, I think *that* is the place to tweak, but we should still write the error to the errorStream here